internal/auth/authController.go

```diff
@@ -0,0 +1,74 @@
+package auth
+
+import (
+	"eventapi/internal/database"
+	"eventapi/internal/session"
+	"fmt"
+
+	"github.com/gofiber/fiber/v2"
+)
+
+const userInfoURL = "https://www.googleapis.com/oauth2/v2/userinfo"
+
+var us *database.UserService
```

Comment:
I don't have a ton of experience with handling services with on-going connections but I'd wager that using a global variable wouldn't be the ideal approach. Should look if some middleware for fiber can help inject those dependencies

Reply:
General consensus seems to be to use closures instead and initialize everything in main. https://twitter.com/tsenart/status/485920561391239168